lib/console.js

```diff
@@ -33,7 +35,8 @@ function Console(stdout, stderr) {
 }
 
 Console.prototype.log = function() {
-  this._stdout.write(util.format.apply(this, arguments) + '\n');
+  this._stdout.write(util.format.apply(this, arguments).replace(/^/gm,
+          Array(this._group * 2 + 1).join(' ')) + '\n');
```

Comment:
Thank you for reviewing and providing input. I tried to be minimal and not introduce lot of new code and variables to core lib.

I'll update the PR soon based on your comments.

@3y3 For tracking multiple levels of grouping I'd still keep numeric `_group` but generate a fixed prefix variables which would be conditionally concatenated as @petkaantonov suggested. This adds variables and code, but I think it pleases everyone?